pkg/cover/html.go

```diff
@@ -24,40 +24,46 @@ import (
 	"github.com/google/syzkaller/sys/targets"
 )
 
-func fixUpPCs(target string, progs []Prog, coverFilter map[uint32]uint32) []Prog {
-	if coverFilter != nil {
-		for _, prog := range progs {
+func (rg *ReportGenerator) FixupPCs(target *targets.Target, progs []Prog, coverFilter map[uint32]uint32) []Prog {
+	nProgs := make([]Prog, len(progs))
+	copy(nProgs, progs)
+
+	// On arm64 as PLT is enabled by default, .text section is loaded after .plt section,
+	// so there is X bytes offset from module load address for .text section
+	// we need to remove the X bytes offset in order to correct module symbol address.
+	// On some chip, the offset is 0x18 while some newer chip is 0x24,
+	// so for linux, auto detects the offset by subtract module load address from first loaded symbol in module.
+	if target.Arch == targets.ARM64 && target.ModuleLoadOffset != 0 {
+		for i, prog := range nProgs {
 			var nPCs []uint64
 			for _, pc := range prog.PCs {
-				if coverFilter[uint32(pc)] != 0 {
-					nPCs = append(nPCs, pc)
+				if pc < rg.BaseAddr && pc > ((rg.BaseAddr>>32)<<32) {
```

Comment:
What are the semantics of this check? Do we really want to check for "the same aligned 4GB range as the kernel text"? I can't imagine we want to check for precisely that...

Reply:
No needed now, previous introduced for debugging pc != frame.PC